### ISSUE TITLE
X509 certificate does not have a private key.

### ISSUE THREAD

**Issue by kspearrin** [IdentityServer/IdentityServer4#679]
### Issue / Steps to reproduce the problem

1. Generate signing certificate using makecert command from blog post: https://brockallen.com/2015/06/01/makecert-and-creating-ssl-or-signing-certificates/

`makecert -r -pe -n "CN=MyCert" -b 01/01/2015 -e 01/01/2020 -eku 1.3.6.1.5.5.7.3.3 -sky signature -a sha256 -len 4096 -ss my -sr LocalMachine`

2. Add cert on Startup:

`.AddSigningCredential("THE_THUMBPRINT", nameType: NameType.Thumbprint)`

3. Receive error:

```
An exception of type 'System.InvalidOperationException' occurred in IdentityServer4.dll but was not handled in user code

Additional information: X509 certificate does not have a private key.
```

Am I doing something wrong here?

**Comment by leastprivilege** [comment]
It seems the account running identityserver does not have read access to the privater key. You can verify that in the cert MMC

**Comment by kspearrin** [comment]
Thanks. I figured it out.

**Comment by leastprivilege** [comment]
And what was the issue?

---
Dominick Baier




On Thu, Jan 12, 2017 at 6:27 PM +0100, "Kyle Spearrin" <notifications@github.com> wrote:












Closed #679.

**Comment by kspearrin** [comment]
I lied and wasn't using the exact `makecert` command that I posted above. I had some the following:

```
makecert -r -pe -n "CN=MyCert" -b 01/01/2015 -e 01/01/2020 -eku 1.3.6.1.5.5.7.3.3 -sky signature -a sha256 -len 4096 MyCert.cer
```

and then just installed `MyCert.cer` via CCM. Outting the `.cer` with `makecert` this way does not preserve the private key.
